api/src/main/java/org/openmrs/ChemoDosingInstructions.java

```diff
@@ -0,0 +1,171 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License, 
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can 
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under 
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS 
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs;
+
+import java.util.Date;
+import java.util.Locale;
+import java.util.StringTokenizer;
+
+import org.openmrs.api.APIException;
+import org.springframework.validation.Errors;
+import org.springframework.validation.ValidationUtils;
+
+/**
+ * @since 2.4.0
+ */
+public class ChemoDosingInstructions implements DosingInstructions {
+
+	private Double dosageAdjustmentPercentage;
+
+	private String dosageDeliveredTiming;
+
+	private String dosageDilutionInstructions;
+
+	/**
+	 * @see DosingInstructions#getDosingInstructionsAsString(java.util.Locale)
+	 */
+	@Override
+	public String getDosingInstructionsAsString(Locale locale) {
+		StringBuilder dosingInstructions = new StringBuilder();
+
+		// serializing adjustment percentage, timing and dilution instructions to compose dosing instructions
+		if (this.dosageAdjustmentPercentage != null) {
+			dosingInstructions.append("Dosage Adjustment Percentage : ");
+			dosingInstructions.append(this.dosageAdjustmentPercentage);
+			dosingInstructions.append("\n");
```

Comment:
This code can be moved to a single method and reuse that inside other if conditions since its doing the same thing. For example 
```
formatDosingInformation(String label,String value){
return label + " : " + value + "\n";
}
```

and then call this method inside the if conditions.

Reply:
Good idea!

Reply:
Changed!